evm/spec/tables/arithmetic.tex

```diff
@@ -1,4 +1,54 @@
+    \item 1 column storing the frequency of appearance of each value in the range $[0, 2^{16} - 1]$.
+\end{itemize}
+
+\paragraph{Note on $op$:} The opcode column is only used for range-checks. For optimization purposes, we check all arithmetic operations against the cpu table together. To ensure correctness, we also check that the operation's opcode corresponds to its behavior. But range-check is not associated to a unique operation: any operation in the cpu table might require its values to be checked. Thus, the arithmetic table cannot know its opcode in advance: it needs to store the value provided by the cpu table.
+
+\subsubsection{Auxiliary columns}
+The way auxiliary values are leveraged to efficiently check correctness is not trivial, but it is explained in detail in each dedicated file. Overall, five files explain the implementations of the various checks. Refer to:
+\begin{enumerate}
+    \item ``mul.rs'' for details on multiplications.
+    \item ``addcy.rs'' for details on addition, subtraction, ``lt'' and ``gt''. 
+    \item ``modular.rs'' for details on how modular operations are checked. Note that even though ``div'' and ``mod'' are generated and checked in a separate file, they leverage the logic for modular operations described in ``modular.rs''. 
+    \item ``byte'' for details on how ``byte'' is checked.
+    \item ``shift.rs'' for details on how shifts are checked. 
+\end{enumerate}
+
+\paragraph*{Note on ``lt'' and ``gt'':} For ``lt'' and ``gt'', auxiliary columns hold the difference $d$ between the two inputs $x_1, x_2$. We can then treat them similarly to subtractions by ensuring that $x_1 - x_2 = d$ for ``lt'' and $x_2 - x_1 = d$ for ``gt''. An auxiliary column $cy$ is used for the carry in additions and subtractions. In the comparisons case, it holds the overflow flag. Contrary to subtractions, the output of ``lt'' and ``gt'' operations is not $d$ but $cy$. 
+
+\paragraph*{Note on ``div'':} It might be unclear why ``div'' and ``mod'' are dealt with in the same file. 
+
+Given numerator and denominator $n, d$, we compute, like for other modular operations, the quotient $q$ and remainder $\texttt{rem}$: 
+$$div(x_1, x_2) = q * x_2 + \texttt{rem}$$. 
+We then set the associated auxiliary columns to $\texttt{rem}$ and the output to $q$. 
+
+This is why ``div'' is essentially a modulo operation, and can be addressed in almost the same way as ``mod''. The only difference is that in the ``mod'' case, the output is $\texttt{rem}$ and the auxiliary value is $q$.
+
+\paragraph{Note on shifts:} SHR and SHL leverage DIV and MUL respectively for the checks. Indeed, given inputs $s, x$, the output should be $x >> s$ for ``shr'' (resp. $x << s$ for ``shl''). Since shifts are binary operations, we can use the third input columns to store $s_{\texttt{shifted}} = 1 << s$. Then, we can use the ``div'' logic (resp. ``mul'' logic) to ensure that the output is $\frac{x}{s_{\texttt{shifted}}}$ (resp. $x * s_{\texttt{shifted}}$).
```

Comment:
The first sentence confused me a little. Maybe "SHR and SHL are internally constrained as MUL and DIV with shifted operand" sounds better? Non-blocking though